[Ng Ngai Teng Colin] iP

Colin386 opened the pull request.

Colin386 committed:
Level-0 Greet
Add Greet message

Colin386 committed:
Level 1
Greet Echo Exit
Added Echo command

Colin386 committed:
Level-2
Add, List
Added a list class to keep track of task

Colin386 committed:
rename to proper convention

Colin386 committed:
Level 3 Mark as done
Add task class Modify list to use task

Colin386 committed:
make setters and getters be in standard form

Colin386 committed:
Add subclass Deadline Todo Event

Colin386 committed:
Add addTask report
fixed event mistaken as deadline bug

Colin386 committed:
Edit command line testing basic

Colin386 committed:
changed printing formatting of MyList
Modified program testing

Colin386 committed:
Separated method to format due dates for events

Colin386 committed:
Add Comments to program

0xEljh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,17 @@
+    private String atDate;
+
+    public Event(String thing, String date){
+        super(thing);
```
Consider a different variable name; thing seems a bit too vague.

0xEljh commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,17 @@
+
+    public Event(String thing, String date){
+        super(thing);
+        final int AT_LENGTH = 3;
```
Consider avoiding the use of AT_LENGTH; it appears to be a "magic number".

0xEljh commented on src/main/java/MyList.java:
```diff
@@ -1,17 +1,52 @@
+        switch (itemType) {
+
+        case ("todo"):
+            commandLength = 4;
```
Consider a cleaner way to implement commandLength

HengFuYuen commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,82 @@
+     *
+     * @return void
+     */
+    public static void welcomeMessage() {
```
Hi! As this is a method, perhaps you can consider adding a verb like show or print to the method name. You can also consider this for your other methods.

HengFuYuen commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,111 @@
+     *
+     * @return number of items in the list
+     */
+    public int getSize(){
```
Perhaps you can consider leaving a whitespace between ) and { to make your code more consistent and you can also consider this suggestion for other methods as well.

0xEljh reviewed (commented):
Overall, consider streamlining the process of adding a new task such that no "magic numbers" need to be involved. Shortening the function could also be considered as it is getting long.

HengFuYuen commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,39 @@
+     * @return String containing the Deadline tag, the task name and the due date
+     */
+    public String toString() {
+
```
Hi! Maybe you can consider removing these additional blank lines.

HengFuYuen reviewed (commented):
The coding standard looks good!
Perhaps some minor changes would make it better!

Colin386 committed:
Add UnrecognisedCommandException
Allows for program to only add in a fixed amount of items such as Todo, Deadline and Event.

Colin386 committed:
Add extraction methods
New methods allow for easier separation of command information such as date and activity from Deadline and Event

Colin386 committed:
Add Exceptions for handling missing information from users

Colin386 committed:
Add Exception Handling for Done
Function will now handle incorrect numbers entered for the done function

Colin386 committed:
Merge branch 'branch-Level-5'

Colin386 committed:
Packaged programmes into Activities and Exceptions

Colin386 committed:
Merge branch 'branch-A-Packages'

Colin386 committed:
Add Delete function

Colin386 committed:
Changed from Array to ArrayList in MyList

Colin386 committed:
Add Saving ability
Program can now save the task list into a text file
Automatically creates the directory and file wherever the code is run

Colin386 committed:
Add File Reading
Program can read txt file to obtain previously saved data from computer

Colin386 committed:
Merge branch 'branch-Level-6'

Colin386 committed:
ignore duke.txt text files

Colin386 committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	.gitignore

Colin386 committed:
Resolve change in data structure
Modification of certain functions to allow for support of ArrayList instead of array

Colin386 committed:
Fixed program saving
Program now saves every time there is a change to the activity list

esigalas commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,246 @@
+    //private int size;
+
+    /**new array list implementation*/
+    private ArrayList<Task> things;
```
maybe the name of that variable can be a bit more descriptive

esigalas commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,246 @@
+    /**
+     * Prints out the list of items recorded in a neat format
+     */
+    public void printList(){
```
can add white space between ')' and '{' for consistency

esigalas commented on src/main/java/FileSaver.java:
```diff
@@ -0,0 +1,193 @@
+                    Files.createDirectory(dukeInfoFolder);
+            }
+            //Files.deleteIfExists(dukeInfoLocation);
+            //dukeInfoLocation = Files.createFile(dukeInfoLocation);
```
consider removing unused/commented out code

esigalas commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,246 @@
+     *
+     * @param itemInfo String containing information to produce the required task
+     */
+    public void addItem(String itemInfo)  {
```
consider removing one white space for consistency

esigalas commented on src/main/java/MyList.java:
```diff
@@ -0,0 +1,246 @@
+import main.java.activity.Event;
+import main.java.activity.Task;
+import main.java.activity.ToDo;
+import main.java.dukeExceptions.*;
```
Consider importing classes explicitly!

Colin386 committed:
Add User Interface class
Program operation now has a separate User Interface Class for handling user interactions

Colin386 committed:
Refactor command processor
UserInterface is the UI class
MyList is the task list class
Parser makes sense of the user commands
FileSaver deals with file loading from computer

Colin386 committed:
Add Date Time support
Program now only records a deadline or an event if a proper date for the activity is provided
Known issues: if duke.txt file contains invalid dates, all other information will not be recorded.

Colin386 committed:
Resolve handling of incorrect dates in txt files
Program will ignore entries with incorrect dates but will still process those with correct dates located later in the file

Colin386 committed:
Modify list function to include date parameter
Resolved bug of null pointer returned when using retrieveDate method in Event
When list function is given a date, it will return return all events and deadlines that fall on that date
Numbering of events still follows standard list

Colin386 committed:
Add Find
Find command will accept a user search query and will return all activities that contain the user word input
Find command will number activities according to their original numbering system

Colin386 committed:
Merge pull request #1 from Colin386/branch-Level-8

Branch level 8

Colin386 committed:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/Parser.java

Colin386 committed:
Merge pull request #2 from Colin386/branch-Level-9

Branch level 9

Colin386 committed:
Add comments for activity package
Activity package functions now contain all user documented functions

Colin386 committed:
Add comments for Exceptions

(cherry picked from commit 6e9e7c445e31388bd5c0544ce0b5872c46716dbd)

Colin386 committed:
Add comments for Exceptions

Colin386 committed:
Add comments for all classes

Colin386 committed:
Add comments for all classes

(cherry picked from commit a6acf4c11e47cd24ca6581f2bb456d9f73398e70)

Colin386 committed:
Add basic user guide

Colin386 committed:
Set theme jekyll-theme-modernist

Colin386 committed:
Set theme jekyll-theme-leap-day

Colin386 committed:
Corrected minor formatting issue

Colin386 committed:
Merge branch 'master' of https://github.com/Colin386/ip

Colin386 committed:
Changed formatting of user guide again

Colin386 committed:
Add spaces in headers

Colin386 committed:
Set theme jekyll-theme-modernist

Colin386 committed:
Merge pull request #3 from Colin386/branch-A-JavaDoc

Branch a java doc

Colin386 committed:
Add DateParserClass

Allows the program to accept one date format from the user and print out another date format later

Colin386 committed:
Fix bug in time printing

Colin386 committed:
Fix bug with file saver
Program handle issues with corrupted data.txt file
Update to Readme.md

Colin386 committed:
Add text-ui-test
Fixed minor bugs in formatting

Colin386 committed:
Add txt files for test cases

Colin386 committed:
Merge pull request #4 from Colin386/branch-textUI

Branch text ui
